smart-py/marketplace.py

```diff
@@ -53,12 +59,70 @@ def cancel_swap(self, params):
     def update_fee(self, params):
         sp.verify(sp.sender == self.data.manager)
         self.data.fee = params
-        
+
     @sp.entry_point
-    def update_manager(self, params):
+    def update_fee_account(self, params):
         sp.verify(sp.sender == self.data.manager)
+        self.data.fee_account= params
+
+    @sp.entry_point
+    def update_manager(self, params):
+        sp.if sp.now - self.data.last_ping > 100 * 24 * 3600:
+            sp.verify(self.data.secondary_accounts.contains(sp.sender))
+        sp.else:
+            sp.verify(sp.sender == self.data.manager)
+
         self.data.manager = params
-        
+        self.data.secondary_accounts.add(params)
+
+    @sp.entry_point
+    def add_secondary_account(self, params):
+        sp.verify(sp.sender == self.data.manager)
+        self.data.secondary_accounts.add(params)
+
+    @sp.entry_point
+    def remove_secondary_account(self, params):
+        sp.verify(sp.sender == self.data.manager)
+        self.data.secondary_accounts.remove(params)
```

Comment:
I'm not familiar with this data structure; if the manager doesn't exist in the set, will there be any issues trying to remove it?

Reply:
I don't think so. Worst case the operation will fail.

Reply:
https://smartpy.io/reference.html#_sets